op-chain-ops/cmd/celo-migrate/ancients.go

```diff
@@ -141,11 +141,11 @@ func readAncientBlocks(ctx context.Context, freezer *rawdb.Freezer, startBlock,
 	return nil
 }
 
-func transformBlocks(ctx context.Context, in <-chan RLPBlockRange, out chan<- RLPBlockRange) error {
+func transformBlocks(ctx context.Context, in <-chan RLPBlockRange, out chan<- RLPBlockRange, startBlock uint64) error {
```

Comment:
More a theoretical question, what happens if there are no ancients in the db when this is run. Wouldn't this underflow?

Reply:
Yes, but it's actually okay because it overflows again in the actual check. I added a comment clarifying this

Reply:
@palango I don't think the script works if there are no ancients, there are a number of places.

Checked with no-ancients
```
./bin/celo-migrate full --old-db ../../chaindatas/alfajores_small/celo/chaindata --new-db alfajores_small_migrated --deploy-config ../../celo-l2-node-docker-compose/envs/alfajores/migration-config/config.json  --l1-deployments ../../celo-l2-node-docker-compose/envs/alfajores/migration-config/deployment-l1.json --l2-allocs ../../celo-l2-node-docker-compose/envs/alfajores/migration-config/l2-allocs.json --l1-rpc https://ethereum-holesky-rpc.publicnode.com --outfile.rollup-config out-rollup.json --outfile.genesis out-genesis.json --migration-block-number 71458 --migration-block-time `date +%s`
INFO [12-18|15:52:50.823] Full Migration Started                   oldDBPath=../../chaindatas/alfajores_small/celo/chaindata newDBPath=alfajores_small_migrated
INFO [12-18|15:52:50.823] Allocated cache and file handles         database=../../chaindatas/alfajores_small/celo/chaindata cache=1024.00MiB handles=60 readonly=true
INFO [12-18|15:52:50.827] Using LevelDB as the backing database
INFO [12-18|15:52:50.830] Source db is synced to correct height    head=71457 migrationBlock=71458
INFO [12-18|15:52:50.830] Pre-Migration Started                    oldDBPath=../../chaindatas/alfajores_small/celo/chaindata newDBPath=alfajores_small_migrated batchSize=50000 memoryLimit=7500
INFO [12-18|15:52:50.832] Copying files from old database (excluding ancients) process=non-ancients
INFO [12-18|15:52:50.841] Opened ancient database                  database=../../chaindatas/alfajores_small/celo/chaindata/ancient readonly=false
INFO [12-18|15:52:50.882] Running rsync command                    command="/usr/bin/rsync -v -a --exclude=ancient --checksum --delete --progress ../../chaindatas/alfajores_small/celo/chaindata/ alfajores_small_migrated"
INFO [12-18|15:52:50.885] Opened ancient database                  database=alfajores_small_migrated/ancient readonly=false
INFO [12-18|15:52:50.885] Ancient Block Migration Skipped          process=ancients     ancientsInOldDB=0 ancientsInNewDB=0
building file list ...
INFO [12-18|15:52:50.902] TIMER                                    process=ancients     duration=69.081958ms
INFO [12-18|15:52:50.902] Using leveldb as the backing database
INFO [12-18|15:52:50.902] Allocated cache and file handles         database=../../chaindatas/alfajores_small/celo/chaindata cache=1024.00MiB handles=60 readonly=true
INFO [12-18|15:52:50.904] Using LevelDB as the backing database
INFO [12-18|15:52:50.904] Found legacy ancient chain path          location=../../chaindatas/alfajores_small/celo/chaindata/ancient
INFO [12-18|15:52:50.904] Opened ancient database                  database=../../chaindatas/alfajores_small/celo/chaindata/ancient readonly=true
INFO [12-18|15:52:50.907] TIMER                                    process=getStrayAncientBlocks duration=5.420459ms
panic: runtime error: makeslice: cap out of range

goroutine 4 [running]:
github.com/ethereum/go-ethereum/core/rawdb.ReadAllHashesInRange({0x106006378, 0x140003b28d0}, 0x104c01f01?, 0xffffffffffffffff)
	/Users/pierspowlesland/go/pkg/mod/github.com/celo-org/op-geth@v1.101408.1-0.20241209113131-e6499495c176/core/rawdb/accessors_chain.go:93 +0x98
main.getStrayAncientBlocks({0x16bbfe3f1, 0x2f})
	/Users/pierspowlesland/projects/optimism/op-chain-ops/cmd/celo-migrate/ancients.go:293 +0x234
main.runPreMigration.func1()
	/Users/pierspowlesland/projects/optimism/op-chain-ops/cmd/celo-migrate/main.go:301 +0xf4
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/Users/pierspowlesland/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x58
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
	/Users/pierspowlesland/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x98
14 files to consider
./
```

Reply:
But I don't think we should be fixing this, we are always going to have ancients